src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -1821,73 +1850,103 @@ public static long Min(this IEnumerable<long> source)
         public static float Min(this IEnumerable<float> source)
         {
             if (source == null) throw Error.ArgumentNull("source");
-            float value = 0;
-            bool hasValue = false;
-            foreach (float x in source)
+            using (IEnumerator<float> e = source.GetEnumerator())
             {
-                if (hasValue)
+                if (!e.MoveNext()) throw Error.NoElements();
+                float value = e.Current;
```

Comment:
I don't check if the first element of an enumerable `IsNaN`, because that adds a test where the current code does not. Such a check could be easily added as well.